### PR TITLE
Adding origin to NVTs

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

Adding the Severity Origin to the NVT Details.
* e.g. "Third Party"

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gsa/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

**Comment by y0urself** [comment]
origin

File: gsa/src/gmp/models/nvt.js [hunk 1]
```diff
@@ -148,7 +148,14 @@ class Nvt extends Info {
 
     delete ret.refs;
 
-    ret.severity = parseSeverity(ret.cvss_base);
+    if (isDefined(ret.severities)) {
+      const {severities} = ret;
+      const {severity} = severities;
+      ret.severity = parseSeverity(severity.score / 10);
+      ret.severity_origin = parseText(severity.origin);
```
Comment (swaterkamp) [inline gsa/src/gmp/models/nvt.js:155]: Please use camelCase for variables in GSA
```suggestion
      ret.severityOrigin = parseText(severity.origin);
```

File: gsa/src/web/pages/nvts/details.js [hunk 2]
```diff
@@ -43,7 +45,7 @@ import Solution from './solution';
 import Pre from './preformatted';
 
 const NvtDetails = ({entity, links = true}) => {
-  const {tags = {}, severity, qod, family, solution} = entity;
+  const {tags = {}, severity, qod, family, solution, severity_origin} = entity;
```
Comment (swaterkamp) [inline gsa/src/web/pages/nvts/details.js:48]:
```suggestion
  const {tags = {}, severity, qod, family, solution, severityOrigin} = entity;
```

File: gsa/src/web/pages/nvts/details.js [hunk 3]
```diff
@@ -76,6 +78,10 @@ const NvtDetails = ({entity, links = true}) => {
                   </TableData>
                 </TableRow>
               )}
+            <TableRow>
+              <TableData>{_('CVSS Origin')}</TableData>
+              <TableData>{na(severity_origin)}</TableData>
```
Comment (swaterkamp) [inline gsa/src/web/pages/nvts/details.js:83]:
```suggestion
              <TableData>{na(severityOrigin)}</TableData>
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Create alerts in task dialog via graphql [#2425](https://github.com/greenbone/gsa/pull/2425) 
 - Added missing fields for getScanners query and parseObject() for scanner model [#2301](https://github.com/greenbone/gsa/pull/2301)
 - Added the CVSS v3.1 BaseScore calculator to the `/cvsscalculator` page in the Help section. [#2536](https://github.com/greenbone/gsa/pull/2536)
+- Added CVSS Origin to NVT Details Page [#2588](https://github.com/greenbone/gsa/pull/2588)
```
Comment (swaterkamp) [inline CHANGELOG.md:13]: Please add the latest changes to the top of the list

File: gsa/src/web/pages/nvts/details.js (hunk 3 above)
Comment (y0urself) [reply gsa/src/web/pages/nvts/details.js:83]: Sure ... 👍

File: CHANGELOG.md (hunk 4 above)
Comment (swaterkamp) [reply CHANGELOG.md:13]: Same for line 12, I guess, so while your are at it... ;)

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,9 +7,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## [21.04] - unreleased
 
 ### Added
+- Added CVSS Origin to NVT Details Page [#2588](https://github.com/greenbone/gsa/pull/2588)
```
Comment (swaterkamp) [inline CHANGELOG.md:10]:
```suggestion
- Added CVSS origin to NVT detailspage [#2588](https://github.com/greenbone/gsa/pull/2588)
```

**Review by swaterkamp — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,9 +7,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## [21.04] - unreleased
 
 ### Added
+- Added CVSS origin to NVT detailspage [#2588](https://github.com/greenbone/gsa/pull/2588)
```
Comment (swaterkamp) [inline CHANGELOG.md:10]:
```suggestion
- Added CVSS origin to NVT details [#2588](https://github.com/greenbone/gsa/pull/2588)
```

**Review by swaterkamp — APPROVED** [review]

**Review by sarahd93 — CHANGES_REQUESTED** [review]
We need to adjust the tests for the nvt model. There is already a test "should parse severity" that can be renamed to "should parse severity from cvss_base". And we need a new test "should parse severity and severity origin" to make sure that works as well. If I understand it correctly in case we have severities and cvss_base defined we want to use the severities and not the cvss_base so the test should cover that as well.

File: gsa/src/gmp/models/__tests__/nvt.js [hunk 7]
```diff
@@ -154,6 +154,43 @@ describe('nvt Model tests', () => {
...
   });
 
+  test('should parse severity and origin from severities', () => {
+    const nvt1 = Nvt.fromElement({
+      severities: {
+        severity: {
+          score: 94,
+          origin: 'Vendor',
+        },
+      },
```
Comment (sarahd93) [inline gsa/src/gmp/models/__tests__/nvt.js:164]:
```suggestion
      },
      cvss_base: '7.5'
```

Comment (sarahd93) [reply gsa/src/gmp/models/__tests__/nvt.js:164]: We should add something like this to make sure that the cvss_base is ignored and the severity from severities is used. That is how it is supposed to work, right, we only use the cvss_base as a fall back?

File: gsa/src/gmp/models/__tests__/nvt.js [hunk 8]
```diff
@@ -154,6 +154,43 @@ describe('nvt Model tests', () => {
...
+          score: 74,
+          origin: 'Greenbone',
+        },
+      },
+    });
+    const nvt3 = Nvt.fromElement({
+      severities: {
+        severity: {
+          score: 10,
+          origin: 'Third Party',
```
Comment (sarahd93) [inline gsa/src/gmp/models/__tests__/nvt.js:178]: Can origin be undefined or empty? We should have one nvt in the test where we don't have an origin to make sure we can handle that correctly.

**Review by sarahd93 — CHANGES_REQUESTED** [review]

**Review by sarahd93 — APPROVED** [review]

**Pull request merged**